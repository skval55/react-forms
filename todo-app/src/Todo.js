import React from "react";

const Todo = ({ id, todo, handleDelete }) => {
  return (
    <li>
      {todo}
      <button onClick={() => handleDelete(id)}>X</button>
    </li>
  );
};

export default Todo;
