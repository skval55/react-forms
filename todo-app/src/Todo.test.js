import { render, screen } from "@testing-library/react";
import Todo from "./Todo";

// smoke test
test("renders without crashing", () => {
  render(
    <Todo
      key="{task.id}"
      id="{task.id}"
      todo="{task.todo}"
      handleDelete={() => console.log("test")}
    />
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Todo
      key="{task.id}"
      id="{task.id}"
      todo="{task.todo}"
      handleDelete={() => console.log("test")}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
