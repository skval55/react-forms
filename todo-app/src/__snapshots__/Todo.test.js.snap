// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    {task.todo}
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
